shared/js/media/media_frame.js

```diff
@@ -137,6 +139,26 @@ MediaFrame.prototype.displayImage = function displayImage(blob,
   // It would be nice if users of this module could override this
   // background color.
   this.image.style.backgroundColor = '#222';
+  this.image.setAttribute('role', 'img');
+
+  this._localize = function _localize() {
+    var label;
+    var orientation = rotation === 90 || rotation === 270 ?
+      'orientationPortrait' : 'orientationLandscape';
+    if (timestamp) {
+      var locale_entry = navigator.mozL10n.get('imageDescription',
+        { orientation: navigator.mozL10n.get(orientation) });
+      label = MediaUtils.formatDate(timestamp, locale_entry);
+    } else {
+      label = navigator.mozL10n.get('imageDescriptionNoTimestamp',
+        { orientation: navigator.mozL10n.get(orientation) });
+    }
+    this.image.setAttribute('aria-label', label);
+  };
```

Comment:
This is shared code you're modifying here, also used by the Camera app, so you'll want to modify that app to be sure these new l10n strings are defined for that app as well.

Also, since it is shared, you should probably write this code to be robust in the case that the get() calls return null.  Line 151 should be protected by if (locale_entry) and line 156 should be protected by if (label), I think.